Adding new tools: Dig and Dnsenum (Issues #26 #27)

mayara-santos01 opened the pull request:
Fix #26 
Fix #27 

### Description

This PR adds the tools Dig and Dnsenum, in order to fix the issues #26 and #27. I executed some tests to check if the changes made were correct and attached screenshots related to them. I am also adding the installation of iodine in the Dockerfile. Please check if this PR can be merged or if there is anything that I should change. 

![Schermata del 2025-02-19 16-25-08](https://github.com/user-attachments/assets/9ca0aefe-9c3c-400d-92b0-c661eb64df92)
![Schermata del 2025-02-19 16-16-28](https://github.com/user-attachments/assets/c019455e-a87c-4c43-8f65-48cd66bc36ba)
![Schermata del 2025-02-19 16-24-21](https://github.com/user-attachments/assets/1816e621-ac14-4a94-90db-efd774100959)

mayara-santos01 committed:
Update CHANGELOG.md

mayara-santos01 committed:
Update Dockerfile

mayara-santos01 committed:
Update database.sql

mayara-santos01 committed:
Update Dockerfile

mayara-santos01 committed:
Update Dockerfile

gildasio commented on src/assets/database.sql:
```diff
@@ -444,7 +482,8 @@ INSERT INTO `tools` (`id`, `name`, `fullname`, `categories`, `description`, `sit
 INSERT INTO `tools` (`id`, `name`, `fullname`, `categories`, `description`, `site`, `github`, `released`, `avatar`, `cmd`, `target`, `resume`, `category`, `category2`, `solution`) VALUES (23, 'traceroute', 'traceroute', 'benign route', 'tracks the route packets taken from an IP network on their way to a given host', 'https://traceroute.sourceforge.net/', NULL, 'Yes', NULL, 'traceroute', NULL, NUL ...
 INSERT INTO `tools` (`id`, `name`, `fullname`, `categories`, `description`, `site`, `github`, `released`, `avatar`, `cmd`, `target`, `resume`, `category`, `category2`, `solution`) VALUES (24, 'h2t', 'HTTP Hardening Tool', 'benign vulnerability-analysis', 'h2t (HTTP Hardening Tool) scans a website and suggests security headers to apply', 'https://certbahia.pop-ba.rnp.br/projects/h2t/', 'https://gi ...
 INSERT INTO `tools` (`id`, `name`, `fullname`, `categories`, `description`, `site`, `github`, `released`, `avatar`, `cmd`, `target`, `resume`, `category`, `category2`, `solution`) VALUES (25, 'arjun', 'arjun', 'information-gathering web-application', 'HTTP parameter discovery suite', NULL, 'https://github.com/s0md3v/Arjun', 'Yes', 'assets/img/arjun.png', 'arjun', '-u', NULL, 'Information-Gatherin ...
-INSERT INTO `tools` (`id`, `name`, `fullname`, `categories`, `description`, `site`, `github`, `released`, `avatar`, `cmd`, `target`, `resume`, `category`, `category2`, `solution`) VALUES (26, 'sqlmap', 'sqlmap', 'web-application vulnerability-analysis', 'Automatic SQL injection and database takeover tool', 'https://sqlmap.org/', 'https://github.com/sqlmapproject/sqlmap', 'Yes', NULL, 'sqlmap --ba ...
```
Once you are adding a new command (a new `INSERT` statement) you do not need to touch this line.

Actually, touching it this way would break the build process.
Test it ;)

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,44 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+--
+INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `command`, `example`, `sudo`, `category`) VALUES 
+('DNS Server', 'Defines the DNS server which will be used in the command', 'dig @8.8.8.8 example.hackinsdn.ubfa,br', 27, 'input', NULL, NULL, NULL, 'BASIC OPTIONS'),
+('DNS domain to query', 'Set the DNS domain which will be queried', 'domain example.hackinsdn.ufba.br', 27, 'input', 'domain', NULL, NULL, 'BASIC OPTIONS'),
```
Domain to query should be the target parameter of this tool, not an optional one.

gildasio replied:
Note in your screenshot `domain` being on command line on bulk execution page.

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,44 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+-- Populate data for `commands`: dig
+--
+INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `command`, `example`, `sudo`, `category`) VALUES 
+('DNS Server', 'Defines the DNS server which will be used in the command', 'dig @8.8.8.8 example.hackinsdn.ubfa,br', 27, 'input', NULL, NULL, NULL, 'BASIC OPTIONS'),
```
I'm not sure this will work. To set the dns server dig to use it must be preceeded by `@`.

gildasio replied:
Hey, @mayara-santos01 , please take a look at this point. It would be interesting to set DNS server to use while executing `dig` through Secflood, but this option is missed from the `dig` commands. What about to add it?

italovalcy replied:
yes, this was removed. In order to specify the server, one can type in the main "target" input something like "@172.16.50.1 xpto.com". Please let me know what you think.

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,44 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `command`, `example`, `sudo`, `category`) VALUES 
+('DNS Server', 'Defines the DNS server which will be used in the command', 'dig @8.8.8.8 example.hackinsdn.ubfa,br', 27, 'input', NULL, NULL, NULL, 'BASIC OPTIONS'),
+('DNS domain to query', 'Set the DNS domain which will be queried', 'domain example.hackinsdn.ufba.br', 27, 'input', 'domain', NULL, NULL, 'BASIC OPTIONS'),
+('Query class (default: in)', NULL, 'dig example.com -q-class HS', 27, 'input', '-q-class', NULL, NULL, 'BASIC OPTIONS'),
+('Query type (default: a)', NULL, 'dig example.com -q-type MX', 27, 'input', '-q-type', NULL, NULL, 'BASIC OPTIONS'),
```
As far as I can tell there is no such options like `-q-class` and `-q-type`, please review it.

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,44 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Query type (default: a)', NULL, 'dig example.com -q-type MX', 27, 'input', '-q-type', NULL, NULL, 'BASIC OPTIONS'),
+('Query options, multiple choices', NULL, 'dig example.com -q-opt -4 -b 192.168.1.1#53', 27, 'input', NULL, NULL, NULL, 'BASIC COMMANDS'),
+('Display options, multiple choices', NULL, 'dig example.com +short +dnssec +retry=3 +timeout=5', 27, 'input', NULL, NULL, NULL, 'BASIC COMMANDS'),
+('Global DNS options configuration','Affects all queries', 'dig global-d-opt example.com', 27, 'checkbox', 'global-d-opts', NULL, NULL, 'DNS OPTIONS CONFIGURATIONS'),
+('Local DNS options configurations', 'Affects specific queries', 'dig local-d-opt example.com', 27, 'checkbox', 'local-d-opts', NULL, NULL, 'DNS OPTIONS CONFIGURATIONS');
```
Same as the previous comment: from what I know there is no parameters as `global-d-opts` and `local-d-opts` this way, please review it.

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,44 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('DNS domain to query', 'Set the DNS domain which will be queried', 'domain example.hackinsdn.ufba.br', 27, 'input', 'domain', NULL, NULL, 'BASIC OPTIONS'),
+('Query class (default: in)', NULL, 'dig example.com -q-class HS', 27, 'input', '-q-class', NULL, NULL, 'BASIC OPTIONS'),
+('Query type (default: a)', NULL, 'dig example.com -q-type MX', 27, 'input', '-q-type', NULL, NULL, 'BASIC OPTIONS'),
+('Query options, multiple choices', NULL, 'dig example.com -q-opt -4 -b 192.168.1.1#53', 27, 'input', NULL, NULL, NULL, 'BASIC COMMANDS'),
+('Display options, multiple choices', NULL, 'dig example.com +short +dnssec +retry=3 +timeout=5', 27, 'input', NULL, NULL, NULL, 'BASIC COMMANDS'),
```
I think it would be great to have those display options as pre-defined as the other parameters as well. At least the most common used.

gildasio replied:
@mayara-santos01 you just removed those options, but it would be better to add some once they are very used, like `+short`. You can add them as `checkbox` type.

italovalcy replied:
Added in commit [0176136](https://github.com/hackinsdn/secflood/pull/60/commits/0176136d185c9be196a2aaa12eda33b394dc2e98)

gildasio reviewed (changes requested):
Hi @mayara-santos01 , very nice to see how this is evolving!!

I made a review, please see my comments there. :)

Sorry but now I can't review these `dnsenum` parameters.
What I recommend is to keep opening different pull requests for different improvements (one for adding `dig` and another one to add `dnsenum). This way the review can be more focused and simple.

mayara-santos01 committed:
Update database.sql

mayara-santos01 commented:
@gildasio thanks for you feedback! I made some changes according to your recommendations, but i didn't really understand this one and what I should do based on it:

> Once you are adding a new command (a new `INSERT` statement) you do not need to touch this line.
> 
> Actually, touching it this way would break the build process. Test it ;)

gildasio commented:
> i didn't really understand this one and what I should do based on it:
> 
> > Once you are adding a new command (a new `INSERT` statement) you do not need to touch this line.
> > 
> > Actually, touching it this way would break the build process. Test it ;)

Hi @mayara-santos01 what I mean is that doing such change you broke the execution because a statement line should ends with `;` not `,`. You can try execute the hole file in sqlite to catch the error.

mayara-santos01 committed:
Update database.sql

mayara-santos01 committed:
Merge branch 'hackinsdn:main' into main

mayara-santos01 commented:
> Hi @mayara-santos01 what I mean is that doing such change you broke the execution because a statement line should ends with ; not ,. You can try execute the hole file in sqlite to catch the error.

I see, I made this mistake because I got a bit confused about the structure of SQL tables, I just updated the code and corrected it.

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,54 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('addr/port (opt)', 'Bind to source address/port', 'dig -b 10.0.0.1#80', 27, 'input', '-b', NULL, NULL, 'QUERY OPTIONS'), 
+('Query Class', 'The most commonly used class is IN (Internet), which is the default for most DNS queries. However, there are other classes like CH (Chaos) and HS (Hesiod) which are less used', 'dig -c HS example.com', 27, 'input', '-c', NULL, NULL, 'QUERY OPTIONS'),
+('Port Number', 'Specifies the port number for the query', 'dig -p 80 example.com', 27, 'input', '-p', NULL, NULL, 'QUERY OPTIONS'),
+('Query Name', 'Specifies the query name for the DNS lookup', 'dig -q example.com', 27, 'input', '-q', NULL, NULL, 'QUERY OPTIONS'),
```
Hey @mayara-santos01 , as I can see you can set `-q` to be the `dig`'s `target` option when adding the tool downthere.

italovalcy replied:
yes, that is correct. We can even provide multiple query names when filling the target input. I will remove this option.

italovalcy replied:
this was removed.

gildasio reviewed (changes requested):
@mayara-santos01 sorry for the delay, for better later than never haha

Just pointed out some interesting improvements you can do in this job.

Good work as well! :)

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,54 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('addr/port (opt)', 'Bind to source address/port', 'dig -b 10.0.0.1#80', 27, 'input', '-b', NULL, NULL, 'QUERY OPTIONS'), 
+('Query Class', 'The most commonly used class is IN (Internet), which is the default for most DNS queries. However, there are other classes like CH (Chaos) and HS (Hesiod) which are less used', 'dig -c HS example.com', 27, 'input', '-c', NULL, NULL, 'QUERY OPTIONS'),
+('Port Number', 'Specifies the port number for the query', 'dig -p 80 example.com', 27, 'input', '-p', NULL, NULL, 'QUERY OPTIONS'),
+('Query Name', 'Specifies the query name for the DNS lookup', 'dig -q example.com', 27, 'input', '-q', NULL, NULL, 'QUERY OPTIONS'),
```
```suggestion
```

italovalcy committed:
Update src/assets/database.sql

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Query Class', 'The most commonly used class is IN (Internet), which is the default for most DNS queries. However, there are other classes like CH (Chaos) and HS (Hesiod) which are less used', 'dig -c HS example.com', 27, 'input', '-c', NULL, NULL, 'QUERY OPTIONS'),
+('Port Number', 'Specifies the port number for the query', 'dig -p 80 example.com', 27, 'input', '-p', NULL, NULL, 'QUERY OPTIONS'),
+('Query Type', 'Specifies the query type', 'dig -t MX example.com', 27, 'input', '-t', NULL, NULL, 'QUERY OPTIONS'),
+('Reverse Lookup', 'Set IP address which will be used for reverse lookup', 'dig -x 10.0.0.1', 27, 'input', '-x', NULL, NULL, 'QUERY OPTIONS'),
```
```suggestion
('Reverse Lookup', 'Prepare the query name to be used for reverse lookup (in-addr.arpa)', 'dig -x 10.0.0.1', 27, 'checkbox', '-x', NULL, NULL, 'QUERY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Port Number', 'Specifies the port number for the query', 'dig -p 80 example.com', 27, 'input', '-p', NULL, NULL, 'QUERY OPTIONS'),
+('Query Type', 'Specifies the query type', 'dig -t MX example.com', 27, 'input', '-t', NULL, NULL, 'QUERY OPTIONS'),
+('Reverse Lookup', 'Set IP address which will be used for reverse lookup', 'dig -x 10.0.0.1', 27, 'input', '-x', NULL, NULL, 'QUERY OPTIONS'),
+('Display Flags', 'Set or Clear All Display Flags', 'dig +[no]all example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Display Flags', 'Set All Display Flags', 'dig +all example.com', 27, 'checkbox', '+all', NULL, NULL, 'DISPLAY OPTIONS'),
('Do not Display Flags', 'Clear All Display Flags', 'dig +noall example.com', 27, 'checkbox', '+noall', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Query Type', 'Specifies the query type', 'dig -t MX example.com', 27, 'input', '-t', NULL, NULL, 'QUERY OPTIONS'),
+('Reverse Lookup', 'Set IP address which will be used for reverse lookup', 'dig -x 10.0.0.1', 27, 'input', '-x', NULL, NULL, 'QUERY OPTIONS'),
+('Display Flags', 'Set or Clear All Display Flags', 'dig +[no]all example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Answer Section', 'Control Display of Answer Section', 'dig +[no]answer example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Show Answer Section', 'Control Display of Answer Section', 'dig +answer example.com', 27, 'checkbox', '+answer', NULL, NULL, 'DISPLAY OPTIONS'),
('Hide Answer Section', 'Control Display of Answer Section', 'dig +noanswer example.com', 27, 'checkbox', '+noanswer', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Reverse Lookup', 'Set IP address which will be used for reverse lookup', 'dig -x 10.0.0.1', 27, 'input', '-x', NULL, NULL, 'QUERY OPTIONS'),
+('Display Flags', 'Set or Clear All Display Flags', 'dig +[no]all example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Answer Section', 'Control Display of Answer Section', 'dig +[no]answer example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Authority Section', 'Control Display of Authority Section', 'dig +[no]authority example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Show Authority Section', 'Control Display of Authority Section', 'dig +authority example.com', 27, 'checkbox', '+authority', NULL, NULL, 'DISPLAY OPTIONS'),
('Hide Authority Section', 'Control Display of Authority Section', 'dig +noauthority example.com', 27, 'checkbox', '+noauthority', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Display Flags', 'Set or Clear All Display Flags', 'dig +[no]all example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Answer Section', 'Control Display of Answer Section', 'dig +[no]answer example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Authority Section', 'Control Display of Authority Section', 'dig +[no]authority example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Question Section', 'Control Display of Question Section', 'dig +[no]question example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Show Question Section', 'Control Display of Question Section', 'dig +question example.com', 27, 'checkbox', '+question', NULL, NULL, 'DISPLAY OPTIONS'),
('Hide Question Section', 'Control Display of Question Section', 'dig +noquestion example.com', 27, 'checkbox', '+noquestion', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Answer Section', 'Control Display of Answer Section', 'dig +[no]answer example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Authority Section', 'Control Display of Authority Section', 'dig +[no]authority example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Question Section', 'Control Display of Question Section', 'dig +[no]question example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Stats', 'Control Display of Stats', 'dig +[no]stats example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Show Stats', 'Control Display of Statistics', 'dig +stats example.com', 27, 'checkbox', '+stats', NULL, NULL, 'DISPLAY OPTIONS'),
('Hide Stats', 'Control Display of Statistics', 'dig +nostats example.com', 27, 'checkbox', '+nostats', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Authority Section', 'Control Display of Authority Section', 'dig +[no]authority example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Question Section', 'Control Display of Question Section', 'dig +[no]question example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Stats', 'Control Display of Stats', 'dig +[no]stats example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('DNSSEC', 'Request DNSSEC Records', 'dig +[no]dnssec example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
```
```suggestion
('Enable DNSSEC', 'Request DNSSEC records', 'dig +dnssec example.com', 27, 'checkbox', '+dnssec', NULL, NULL, 'QUERY OPTIONS'),
('Disable DNSSEC', 'Do not request DNSSEC records', 'dig +nodnssec example.com', 27, 'checkbox', '+nodnssec', NULL, NULL, 'QUERY OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Display TTL', 'Display Time to Live (TTL) in Records', 'dig +[no]ttlid example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Human-Readble TTL', 'Display TTL in Human-Readable Units', 'dig +[no]ttlunits example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Set Query Timeout', NULL, 'dig +timeout=5 example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS'),
+('Per-Record Comments', 'Control Display of Per-Record Comments', 'dig +[no]rrcomments example.com', 27, 'input', NULL, NULL, NULL, 'DISPLAY OPTIONS');
```
```suggestion
('Enable Trace', 'Trace delegation down from root', 'dig +trace example.com', 27, 'checkbox', '+trace', NULL, NULL, 'QUERY OPTIONS');
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Set --enum switch', 'Shortcut option equivalent to --threads 5 -s 15 -w.',  NULL, 14, 'checkbox', '--enum', NULL, NULL, 'GENERAL OPTIONS'),
+('Skip reverse lookup', 'Skip the reverse lookup (reverse DNS searc) operations.', NULL, 14, 'checkbox', '--noreverse', NULL, NULL, 'GENERAL OPTIONS'),
+('No colors','Disable ANSIColor output, i.e., disables output with colored text.', NULL, 14, 'checkbox', '--nocolor', NULL, NULL, 'GENERAL OPTIONS'),
+('Private ips at the end', 'Show and save private ips at the end of the file domain_ips.txt.', NULL, 14, 'checkbox', '--private', NULL, NULL, 'GENERAL OPTIONS'),
```
```suggestion
('Private IPs at the end', 'Show and save private IPs at the end of the file domain_ips.txt.', NULL, 14, 'checkbox', '--private', NULL, NULL, 'GENERAL OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Private ips at the end', 'Show and save private ips at the end of the file domain_ips.txt.', NULL, 14, 'checkbox', '--private', NULL, NULL, 'GENERAL OPTIONS'),
+('File to save subdomains','Write all valid subdomains to the chosen file.', 'dnsenum --subfile subdomains_obtained.txt example.com', 14, 'input', '--subfile', NULL, NULL, 'GENERAL OPTIONS'),
+('Timeout setting','The tcp and udp timeout values in seconds. Defines he interval of time in which dnsenum waits from an answer from the DNS server (default: 10s).', 'dnsenum -t 10 example.com', 14, 'input', '-t', NULL, NULL, 'GENERAL OPTIONS'),
+('Threads setting', 'The number of threads that will perform different queries.', 'dnsenum --threads 5 example.com', 14, 'checkbox', '--threads', NULL, NULL, 'GENERAL OPTIONS'),
```
```suggestion
('Timeout','The tcp and udp timeout values in seconds. Defines he interval of time in which dnsenum waits from an answer from the DNS server (default: 10s).', 'dnsenum -t 10 example.com', 14, 'input', '-t', NULL, NULL, 'GENERAL OPTIONS'),
('Threads', 'The number of threads that will perform different queries.', 'dnsenum --threads 5 example.com', 14, 'checkbox', '--threads', NULL, NULL, 'GENERAL OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Timeout setting','The tcp and udp timeout values in seconds. Defines he interval of time in which dnsenum waits from an answer from the DNS server (default: 10s).', 'dnsenum -t 10 example.com', 14, 'input', '-t', NULL, NULL, 'GENERAL OPTIONS'),
+('Threads setting', 'The number of threads that will perform different queries.', 'dnsenum --threads 5 example.com', 14, 'checkbox', '--threads', NULL, NULL, 'GENERAL OPTIONS'),
+('Verbose output', 'Be verbose: show all the progress and all the error messages.', NULL, 14, 'checkbox', '-v', NULL, NULL, 'GENERAL OPTIONS'),
+('Google pages scrapping', 'The number of google search pages to process when scraping names, the default is 5 pages, the -s switch must be specified.', 'dnsenum -p 15 - 20 example.com', 14, 'input', '-p', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS'),
+('Subdomains scrapping', 'The maximum number of subdomains that will be scraped from Google (default 15)', 'dnsenum -s 40 example.com', 14, 'input', '-s', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS: SUBDOMAINS'),
```
```suggestion
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Verbose output', 'Be verbose: show all the progress and all the error messages.', NULL, 14, 'checkbox', '-v', NULL, NULL, 'GENERAL OPTIONS'),
+('Google pages scrapping', 'The number of google search pages to process when scraping names, the default is 5 pages, the -s switch must be specified.', 'dnsenum -p 15 - 20 example.com', 14, 'input', '-p', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS'),
+('Subdomains scrapping', 'The maximum number of subdomains that will be scraped from Google (default 15)', 'dnsenum -s 40 example.com', 14, 'input', '-s', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS: SUBDOMAINS'),
+('Choose the brute force file', 'Read subdomains from this file to perform brute force. (Takes priority over default dns.txt)', 'dnsenum -f subdomains.txt example.com', 14, 'input', '-f', NULL, NULL, 'SUBDOMAIN BRUTE FORCE'), 
```
```suggestion
('Choose the brute force file', 'Read subdomains from this file to perform brute force. (Takes priority over default dns.txt)', 'dnsenum -f subdomains.txt example.com', 14, 'input', '-f', NULL, NULL, 'BRUTE FORCE OPTIONS'), 
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Google pages scrapping', 'The number of google search pages to process when scraping names, the default is 5 pages, the -s switch must be specified.', 'dnsenum -p 15 - 20 example.com', 14, 'input', '-p', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS'),
+('Subdomains scrapping', 'The maximum number of subdomains that will be scraped from Google (default 15)', 'dnsenum -s 40 example.com', 14, 'input', '-s', NULL, NULL, 'GOOGLE SCRAPPING OPTIONS: SUBDOMAINS'),
+('Choose the brute force file', 'Read subdomains from this file to perform brute force. (Takes priority over default dns.txt)', 'dnsenum -f subdomains.txt example.com', 14, 'input', '-f', NULL, NULL, 'SUBDOMAIN BRUTE FORCE'), 
+('Update brute force file using all results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u a', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Update brute force file using google scrapping results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u g', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Update brute force file using zonetransfer results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u z', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
```
```suggestion
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Update brute force file using all results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u a', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Update brute force file using google scrapping results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u g', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Update brute force file using zonetransfer results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u z', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Recursion on subdomains', 'Executes a brute force in all discovered subdomains that have an NS record.', NULL, 14, 'checkbox', '-r', NULL, NULL, 'BRUTE FORCE OPTIONS: RECURSION'),
```
```suggestion
('Recursion on subdomains', 'Executes a brute force in all discovered subdomains that have an NS record.', NULL, 14, 'checkbox', '-r', NULL, NULL, 'BRUTE FORCE OPTIONS'),
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Update brute force file using google scrapping results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u g', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Update brute force file using zonetransfer results', 'This command updates the file specified with the -f switch.', NULL, 14, 'checkbox', '-u z', NULL, NULL, 'BRUTE FORCE OPTIONS: UPDATE FILE'),
+('Recursion on subdomains', 'Executes a brute force in all discovered subdomains that have an NS record.', NULL, 14, 'checkbox', '-r', NULL, NULL, 'BRUTE FORCE OPTIONS: RECURSION'),
+('whois queries', 'Perform the whois queries on c class network ranges.**Warning**: this can generate very large netranges and it will take lot of time to perform reverse lookups.', 'dnsenum -w example.com', 14, 'checkbox', '-w', NULL, NULL, 'WHOIS NETRANGE OPTIONS'),
+('whois queries delay', 'The maximum value of seconds to wait between whois queries, the value is defined randomly, default: 3s.', 'dnsenum -w -d 5 example.com', 14, 'input', '-d', NULL, NULL, 'WHOIS NETRANGE OPTIONS: DELAY'),
```
```suggestion
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Recursion on subdomains', 'Executes a brute force in all discovered subdomains that have an NS record.', NULL, 14, 'checkbox', '-r', NULL, NULL, 'BRUTE FORCE OPTIONS: RECURSION'),
+('whois queries', 'Perform the whois queries on c class network ranges.**Warning**: this can generate very large netranges and it will take lot of time to perform reverse lookups.', 'dnsenum -w example.com', 14, 'checkbox', '-w', NULL, NULL, 'WHOIS NETRANGE OPTIONS'),
+('whois queries delay', 'The maximum value of seconds to wait between whois queries, the value is defined randomly, default: 3s.', 'dnsenum -w -d 5 example.com', 14, 'input', '-d', NULL, NULL, 'WHOIS NETRANGE OPTIONS: DELAY'),
+('PTR records exclusion', 'Exclude PTR records that match the regexp expression from reverse lookup results, useful on invalid hostnames. In this sense, when a reverse lookup is made, subdomains with the defined words are not considered.', 'dnsenum -e "(test|local|invalid)" examples.', 14, 'input', '-e', NULL, NULL, 'REVERSE LOOKUP OPTIONS'),
```
```suggestion
```

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,53 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('whois queries', 'Perform the whois queries on c class network ranges.**Warning**: this can generate very large netranges and it will take lot of time to perform reverse lookups.', 'dnsenum -w example.com', 14, 'checkbox', '-w', NULL, NULL, 'WHOIS NETRANGE OPTIONS'),
+('whois queries delay', 'The maximum value of seconds to wait between whois queries, the value is defined randomly, default: 3s.', 'dnsenum -w -d 5 example.com', 14, 'input', '-d', NULL, NULL, 'WHOIS NETRANGE OPTIONS: DELAY'),
+('PTR records exclusion', 'Exclude PTR records that match the regexp expression from reverse lookup results, useful on invalid hostnames. In this sense, when a reverse lookup is made, subdomains with the defined words are not considered.', 'dnsenum -e "(test|local|invalid)" examples.', 14, 'input', '-e', NULL, NULL, 'REVERSE LOOKUP OPTIONS'),
+('Output in XML format (Choose the file)', 'Can be imported in MagicTree (www.gremwell.com).', 'dnsenum -o output.xml example.com', 14, 'input', '-o', NULL, NULL, 'OUTPUT OPTIONS');
```
```suggestion
('Output in XML (Enter the file name)', 'Set the output to a file formatted as XML. Can be imported in MagicTree (www.gremwell.com).', 'dnsenum -o output.xml example.com', 14, 'input', '-o', NULL, NULL, 'OUTPUT OPTIONS');
```

italovalcy reviewed (commented):
Added a few suggestions after running for real in a lab with secflood

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy committed:
Update src/assets/database.sql

italovalcy commented on src/assets/database.sql:
```diff
@@ -366,6 +366,48 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Show Stats', 'Control Display of Statistics', 'dig +stats example.com', 27, 'checkbox', '+stats', NULL, NULL, 'DISPLAY OPTIONS'),
+('Hide Stats', 'Control Display of Statistics', 'dig +nostats example.com', 27, 'checkbox', '+nostats', NULL, NULL, 'DISPLAY OPTIONS'),
+('Enable DNSSEC', 'Request DNSSEC records', 'dig +dnssec example.com', 27, 'checkbox', '+dnssec', NULL, NULL, 'QUERY OPTIONS'),
+('Disable DNSSEC', 'Do not request DNSSEC records', 'dig +nodnssec example.com', 27, 'checkbox', '+nodnssec', NULL, NULL, 'QUERY OPTIONS'),
```
```suggestion
('Disable DNSSEC', 'Do not request DNSSEC records', 'dig +nodnssec example.com', 27, 'checkbox', '+nodnssec', NULL, NULL, 'QUERY OPTIONS'),
('Short answer', 'Display nothing except short form of answer', 'dig +short example.com', 27, 'checkbox', '+short', NULL, NULL, 'DISPLAY OPTIONS'),
```

italovalcy committed:
Update src/assets/database.sql

gildasio commented on src/assets/database.sql:
```diff
@@ -366,6 +366,49 @@ INSERT INTO `commands` (`name`, `description`, `examples`, `tool`, `type`, `comm
+('Query Class', 'The most commonly used class is IN (Internet), which is the default for most DNS queries. However, there are other classes like CH (Chaos) and HS (Hesiod) which are less used', 'dig -c HS example.com', 27, 'input', '-c', NULL, NULL, 'QUERY OPTIONS'),
+('Port Number', 'Specifies the port number for the query', 'dig -p 80 example.com', 27, 'input', '-p', NULL, NULL, 'QUERY OPTIONS'),
+('Query Type', 'Specifies the query type', 'dig -t MX example.com', 27, 'input', '-t', NULL, NULL, 'QUERY OPTIONS'),
+('Reverse Lookup', 'Prepare the query name to be used for reverse lookup (in-addr.arpa)', 'dig -x 10.0.0.1', 27, 'checkbox', '-x', NULL, NULL, 'QUERY OPTIONS'),
```
This option requires a target, but I understand what you did by just using the main command target. But this way, if a user check any other option it would not execute. For example: user set target as any IP address then check this reverse lookup and short answer checkboxes. This way the command would end up as being `dig -x +short IP`, which can't execute.

I possible workaround would be move this option as being the last one for this tool once Secflood select it from database or process it. A better solution would, someway, prioritize it to be the last option set.

But, yet, it can be simpler just set it as `input` option so user need to actually pass an IP address to it.

However, any other way can work better for you. Please, let me know :)

italovalcy replied:
```suggestion
('Reverse Lookup', 'Query a name in reverse lookup format (in-addr.arpa)', 'dig -x 10.0.0.1', 27, 'input', '-x', NULL, NULL, 'QUERY OPTIONS'),
```

gildasio reviewed (changes requested):
Hey, @italovalcy , very nice work this one!!

I put a simple comment right there. Please, take a look. I would appreciate such a change, but if you decide to move on even so, let me know and I can approve it this way.

italovalcy committed:
Update src/assets/database.sql

gildasio reviewed (approved):
Hey @italovalcy , nice!! Everything is fine from my view. Go on :)
